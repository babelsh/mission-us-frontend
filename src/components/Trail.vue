<template>
  <v-container fluid>
    <div>
      <v-layout text-xs-center>
        <v-flex>
          <img src="../images/logo.png" height="150px">
        </v-flex>
      </v-layout>
    </div>
    <p class="title">{{ title }}</p>
    <p class="body-2 grey--text text-truncate">
      {{ abstract }}
    </p>
    <v-layout>
      <v-flex xs12 :text-xs-right="isSmall">
        <v-btn class="primary">SABER MAIS</v-btn>
      </v-flex>
    </v-layout>
  </v-container>
</template>

<script>
export default {
  name: 'trail',
  props: {
    title: {
      type: String,
      default: '',
    },
    abstract: {
      type: String,
      default: '',
    },
    imageSrc: {
      type: String,
      default: '',
    },
  },
  computed: {
    isSmall() {
      return this.$vuetify.breakpoint.name === 'xs'
        || this.$vuetify.breakpoint.name === 'sm';
    },
  },
};
</script>
