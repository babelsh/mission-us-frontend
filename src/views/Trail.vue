<template>
<div>
  <v-container text-xs-center>
    <v-layout>
      <v-flex lg8>
        <img src="../images/logo.png" height="150px">
      </v-flex>
    </v-layout>
  </v-container>
  <v-container>
    <v-layout>
      <v-flex xs12>
        <p class="headline">{{ title }}</p>
      </v-flex>
    </v-layout>
    <v-layout>
      <v-flex xs12>
        <p class="subheading grey--text">
          {{ abstract }}
        </p>
      </v-flex>
    </v-layout>
    <v-layout>
      <v-flex xs7 offset-xs7 offset-lg0>
        <v-btn class="primary">SABER MAIS</v-btn>
      </v-flex>
    </v-layout>
  </v-container>
</div>
</template>

<script>
export default {
  name: 'trail',
  props: {
    title: {
      type: String,
      default: '',
    },
    abstract: {
      type: String,
      default: '',
    },
    imageSrc: {
      type: String,
      default: '',
    } 
  }
}
</script>

<style scoped>

</style>