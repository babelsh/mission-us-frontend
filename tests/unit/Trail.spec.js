import { mount } from '@vue/test-utils';
import Trail from '../../src/views/Trail.vue';

describe('Trail', () => {
  it('works', () => {
    const wrapper = mount(Trail, {
      propsData: {
        title: 'titulo',
      },
    });
    expect(wrapper.props().title).toBe('titulo');
  });
});
