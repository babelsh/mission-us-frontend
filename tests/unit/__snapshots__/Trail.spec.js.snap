// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.test.js renders correctly setting the props 1`] = `
<div>
  <v-container
    text-xs-center=""
  >
    <v-layout>
      <v-flex
        lg8=""
      >
        <img
          height="150px"
          src="../images/logo.png"
        />
      </v-flex>
    </v-layout>
  </v-container>
   
  <v-container>
    <v-layout>
      <v-flex
        xs12=""
      >
        <p
          class="headline"
        >
          Titulo da trilha
        </p>
      </v-flex>
    </v-layout>
     
    <v-layout>
      <v-flex
        xs12=""
      >
        <p
          class="subheading grey--text"
        >
          
          Resumo da trilha
        
        </p>
      </v-flex>
    </v-layout>
     
    <v-layout>
      <v-flex
        offset-lg0=""
        offset-xs7=""
        xs7=""
      >
        <v-btn
          class="primary"
        >
          SABER MAIS
        </v-btn>
      </v-flex>
    </v-layout>
  </v-container>
</div>
`;
